src/fee/gpu/fee.h

```diff
@@ -205,61 +205,58 @@ inline __device__ void lpmv_device(FLOAT *output, int n, FLOAT x) {
     }
 }
 
-inline __device__ int lidx_device(const int l, const int m) {
+inline __device__ int lidx_device(const int i_direction, const int l, const int m) {
     // summation series over l + m => (l*(l+1))/2 + m
-    return ((l * (l + 1)) >> 1) + m;
+    return i_direction*(((NMAX + 1)*(NMAX + 2)) >> 1) + ((l * (l + 1)) >> 1) + m;
```

Comment:
lengendre table is now `((NMAX + 1)*(NMAX + 2)) >> 1)` by `n_dirs` which is why i_direction is now needed.